src/test/java/core/basesyntax/strategy/impl/BalanceOperationHandlerImplTest.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationHandler;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class BalanceOperationHandlerImplTest {
+    private static OperationHandler operationHandler;
+    private Transaction transaction;
+    private Fruit fruit;
+
+    @BeforeClass
+    public static void beforeAll() {
+        operationHandler = new BalanceOperationHandlerImpl();
+    }
+
+    @Before
+    public void setUp() {
+        fruit = new Fruit("mango");
+        transaction = new Transaction(Transaction.Operation.BALANCE, fruit, 52);
+    }
+
+    @Test
+    public void operation_ApplyBalance_ok() {
+        boolean emptyStorage = Storage.storage.isEmpty();
+        Assert.assertTrue(emptyStorage);
+        operationHandler.apply(transaction);
+        int actualSize = Storage.storage.size();
+        Assert.assertEquals(1, actualSize);
+    }
+
+    @Test
+    public void operation_ApplyNegativeBalance_notOk() {
+        transaction.setQuantity(-1);
+        operationHandler.apply(transaction);
+        int expected = 0;
+        int actualSize = Storage.storage.size();
+        Assert.assertNotEquals(expected, actualSize);
```

Comment:
do you expect to have in storage mango with quantity -1?
if so, let's test this
now it looks a bit confusing